src/zenml/log_stores/artifact/artifact_log_exporter.py

```diff
@@ -0,0 +1,272 @@
+                        chunk_index=i,
+                        total_chunks=len(chunks),
+                        id=entry_id,
+                    )
+                )
+
+            return entries
+
+    def _map_severity_to_level(self, severity_text: str) -> LoggingLevels:
+        """Map OTEL severity text to ZenML LoggingLevels enum.
+
+        Args:
+            severity_text: The OTEL severity text.
+
+        Returns:
+            The corresponding LoggingLevels enum value.
+        """
+        if not severity_text:
+            return LoggingLevels.INFO
+
+        severity_upper = severity_text.upper()
+
+        if severity_upper in ["DEBUG", "TRACE"]:
+            return LoggingLevels.DEBUG
+        elif severity_upper in ["INFO", "INFORMATION"]:
+            return LoggingLevels.INFO
+        elif severity_upper in ["WARN", "WARNING"]:
+            return LoggingLevels.WARN
+        elif severity_upper == "ERROR":
+            return LoggingLevels.ERROR
+        elif severity_upper in ["CRITICAL", "FATAL", "EMERGENCY"]:
+            return LoggingLevels.CRITICAL
+        else:
+            return LoggingLevels.INFO
+
+    def _split_to_chunks(self, message: str) -> List[str]:
+        """Split a large message into chunks, handling UTF-8 boundaries.
+
+        Args:
+            message: The message to split.
+
+        Returns:
+            A list of message chunks.
+        """
+        message_bytes = message.encode("utf-8")
+        chunks = []
+        start = 0
+
+        while start < len(message_bytes):
+            end = min(start + DEFAULT_MESSAGE_SIZE, len(message_bytes))
+
+            while end > start:
+                chunk_bytes = message_bytes[start:end]
+                try:
+                    chunk_text = chunk_bytes.decode("utf-8")
+                    chunks.append(chunk_text)
+                    break
+                except UnicodeDecodeError:
+                    end -= 1
+            else:
+                end = min(start + DEFAULT_MESSAGE_SIZE, len(message_bytes))
+                chunks.append(
+                    message_bytes[start:end].decode("utf-8", errors="replace")
+                )
+
+            start = end
+
+        return chunks
+
+    def _write_to_artifact_store(
+        self,
+        log_lines: List[str],
+        log_model: "LogsResponse",
+    ) -> None:
+        """Write log lines to the artifact store.
+
+        Args:
+            log_lines: List of JSON-serialized log entries.
+            log_model: The log model.
+            log_id: The log ID for tracking file counters.
+        """
+        if not log_model.uri or not log_model.artifact_store_id:
+            logger.warning(
+                f"Skipping log write: missing uri or artifact_store_id for log {log_model.id}"
+            )
+            return
+
+        client = Client()
+        artifact_store = _load_artifact_store(
+            log_model.artifact_store_id, client.zen_store
+        )
```

Comment:
This tells me that there isn't an exclusive 1-to-1 relationship between ArtifactLogStore and ArtifactStore, but rather that one ArtifactLogStore can be used as a transparent proxy with any and all ArtifactStore instances that the emitted messages might be targetting. Doesn't that mean that the same ArtifactLogStore instance should be shared by all Stacks in the current process that need a default log store, regardless of ArtifactStore that they might use ?

Reply:
This is now addressed properly.